### PR TITLE
WIP Swift support for Dalite

### PR DESCRIPTION
**Pull request by jbzdak**
Testing instuctions: 

1. Install Dalite on your local system. 
2. Configure Swift for dalite (I deliberately didn't add any instructions here, please check instructions from `Readme.md` if something is not obvious please point it and I'll update that file!)
3. You may store static files using Swift, but that's not strictly neccessary . 
4. Create buch of questions, and add images to them. Verify that the images appear in the Swift Container, and are visible when you preview them on Dalite. 
5. You might also test whether it works when you integrate Dalite with LMS, but I'm not sure if it makes sense (that is: nothing can break here...).

### CODE REVIEW

**Comment by jbzdak** [comment]
@omarkhan This is now ready for review.

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,9 @@
 /log
 /media
 /static
+
+# Pycharm IDE folder
+/.idea
```
Comment (omarkhan) [inline .gitignore]: Nit: I don't think this needs to be in the project `.gitignore`, it can live in your personal `.gitignore`

**Comment by omarkhan** [comment]
@jbzdak I had a quick look, it looks good. I will test it tomorrow morning.

File: README.md [hunk 2]
```diff
@@ -161,6 +161,59 @@ After the tests have finished, you can view the coverage report using:
...
+* Set up following keys in the `settings.py`: 
+
+```
+# Storage to use for user uploads
+DEFAULT_FILE_STORAGE='swift.storage.SwiftStorage'
+SWIFT_CONTAINER_NAME='media'
+
+# Storage to use for static files
+STATICFILES_STORAGE ='swift.storage.StaticSwiftStorage'
+SWIFT_STATIC_CONTAINER_NAME='static'
```
Comment (omarkhan) [inline README.md]: It might be worth pointing out that these containers have to exist, they won't be created automatically

File: README.md [hunk 3]
```diff
@@ -161,6 +161,59 @@ After the tests have finished, you can view the coverage report using:
...
+# This is the url to authentication endpoint of your OpenStack installation 
+SWIFT_AUTH_URL=os.environ["OS_AUTH_URL"]
+# Username for Swift authorization
+SWIFT_USERNAME=os.environ["OS_USERNAME"]
+# Password for Swift authorization
+SWIFT_KEY=os.environ["OS_PASSWORD"]
+# This is the auth version to use, on OVH I used version 2
+SWIFT_AUTH_VERSION=2
+# Tenant ID 
+SWIFT_TENANT_ID=os.environ["OS_TENANT_ID"]
```
Comment (omarkhan) [inline README.md]: You may want to set the region name as well:

```python
SWIFT_EXTRA_OPTIONS = {'region_name': os.environ['OS_REGION_NAME']}
```

File: README.md [hunk 4]
```diff
@@ -161,6 +161,59 @@ After the tests have finished, you can view the coverage report using:
...
+# Storage to use for user uploads
+DEFAULT_FILE_STORAGE='swift.storage.SwiftStorage'
+SWIFT_CONTAINER_NAME='media'
+
+# Storage to use for static files
+STATICFILES_STORAGE ='swift.storage.StaticSwiftStorage'
+SWIFT_STATIC_CONTAINER_NAME='static'
+
+# Credentials
+# This is the url to authentication endpoint of your OpenStack installation
```
Comment (omarkhan) [inline README.md]: Maybe link to the [django-storage-swift](https://github.com/blacktorn/django-storage-swift) docs here for a list of all the settings

File: README.md (hunk 2 above)
Comment (omarkhan) [inline README.md]: The containers will also need to be made public I think

**Comment by omarkhan** [comment]
@jbzdak tested, works perfectly. Apart from my comments above :+1:

File: .gitignore (hunk 1 above)
Comment (jbzdak) [inline .gitignore]: Good point, I wasn't aware of this feature :)

**Comment by jbzdak** [comment]
Note: squasing commits, previous head is: db8577d5789156d546adb91b698dd164cc083756

**Comment by jbzdak** [comment]
Merging on Omar's approval.

**Pull request merged**